tests/check_framework/test_translation.py

```diff
@@ -0,0 +1,39 @@
+from django.core.checks.translation import check_setting_language_code
+from django.test import SimpleTestCase, override_settings
+
+
+class TranslationCheckTests(SimpleTestCase):
+
+    @override_settings(LANGUAGE_CODE="eu")
+    def test_valid_language_code_format_ll_only(self):
+        result = check_setting_language_code(None)
+        self.assertEqual(len(result), 0)
+
+    @override_settings(LANGUAGE_CODE="eü")
+    def test_invalid_language_code_format_ll_only(self):
+        result = check_setting_language_code(None)
+        self.assertEqual(len(result), 1)
+        error = result[0]
+        self.assertEqual(error.id, 'translation.E001')
```

Comment:
I'm thinking it might be better to use the approach in `test_caches.py` rather than repeating the error message here twice.